### PR TITLE
fix(github): Disable caching for GitHub datasources

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Returns GitHub tags/releases fetching to REST and disables caching.

## Context

Addresses performance regressions in the hosted app. Plan is:
- Revert GraphQL and convert to REST. No caching
- Switch to GraphQL, no caching
- Add caching back to GraphQL

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/pr/changelog/github/index.ts [hunk 1]
```diff
@@ -116,26 +115,19 @@ export async function getReleaseList(
   release: ChangeLogRelease
 ): Promise<ChangeLogNotes[]> {
   logger.trace('github.getReleaseList()');
-  // TODO #7154
```
Comment (viceice) [inline lib/workers/repository/update/pr/changelog/github/index.ts:119]: revert

File: lib/modules/datasource/github-tags/index.ts [hunk 2]
```diff
@@ -73,16 +73,28 @@ export class GithubTagsDatasource extends GithubReleasesDatasource {
       : this.getCommit(registryUrl, repo!);
   }
 
+  // istanbul ignore next
```
Comment (viceice) [inline lib/modules/datasource/github-tags/index.ts:76]: why this?

File: lib/modules/datasource/github-tags/index.spec.ts [hunk 3]
```diff
@@ -45,46 +40,69 @@ describe('modules/datasource/github-tags/index', () => {
...
+      httpMock
+        .scope(githubApiHost)
+        .get(`/repos/${packageName}/git/refs/tags/${tag}`)
+        .reply(200, {});
+      const res = await github.getDigest({ packageName: 'some/dep' }, 'v1.2.0');
       expect(res).toBeNull();
     });
 
-    it('supports GHE', async () => {
+    it('supports ghe', async () => {
```
Comment (viceice) [inline lib/modules/datasource/github-tags/index.spec.ts:87]: i think it should stay at `GHE`. 🤔

File: lib/modules/datasource/github-releases/index.ts [hunk 4]
```diff
@@ -239,22 +208,26 @@ export class GithubReleasesDatasource extends Datasource {
    *  - Sanitize the versions if desired (e.g. strip out leading 'v')
    *  - Return a dependency object containing sourceUrl string and releases array
    */
+  // istanbul ignore next
```
Comment (viceice) [inline lib/modules/datasource/github-releases/index.ts:211]: why this?

File: lib/modules/datasource/github-releases/index.ts [hunk 5]
```diff
@@ -163,20 +146,6 @@ export class GithubReleasesDatasource extends Datasource {
     return null;
   }
```
Comment (viceice) [inline lib/modules/datasource/github-releases/index.ts:148]: why we don't cache this?

File: lib/modules/datasource/github-releases/index.ts [hunk 6]
```diff
@@ -67,12 +56,6 @@ export class GithubReleasesDatasource extends Datasource {
...
 
-  @cache({
```
Comment (viceice) [inline lib/modules/datasource/github-releases/index.ts:70]: same, why no longer cache this?

**Review by viceice — COMMENTED** [review]
so we now revert to use rest instead of graphql? can you please update PR description with reason so we can remember later why we did this.

File: lib/modules/datasource/github-releases/index.ts (hunk 6 above)
Comment (rarkins) [reply lib/modules/datasource/github-releases/index.ts:70]: See PR context

File: lib/modules/datasource/github-releases/index.ts (hunk 5 above)
Comment (rarkins) [reply lib/modules/datasource/github-releases/index.ts:148]: See PR context

File: lib/modules/datasource/github-releases/index.ts (hunk 4 above)
Comment (rarkins) [reply lib/modules/datasource/github-releases/index.ts:211]: This PR is intended to be temporary so we can test in the app

File: lib/modules/datasource/github-tags/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/modules/datasource/github-tags/index.ts:76]: This PR is temporary, see context

File: lib/modules/datasource/github-tags/index.spec.ts (hunk 3 above)
Comment (rarkins) [inline lib/modules/datasource/github-tags/index.spec.ts:87]:
```suggestion
    it('supports GHE', async () => {
```

File: lib/workers/repository/update/pr/changelog/github/index.ts [hunk 7]
```diff
@@ -116,26 +115,19 @@ export async function getReleaseList(
   release: ChangeLogRelease
 ): Promise<ChangeLogNotes[]> {
   logger.trace('github.getReleaseList()');
```
Comment (rarkins) [inline lib/workers/repository/update/pr/changelog/github/index.ts:117]:
```suggestion
  logger.trace('github.getReleaseList()');
  // TODO #7154
```

**Comment by rarkins** [comment]
I note that we still have the link header-based pagination which will cause some private repos to fail:

```
         "https://api.github.com/repos/nodejs/node (GET,200)": 1,
         "https://api.github.com/repos/nodejs/node/git/blobs/a107bc9c748cf503f04416f780ed362068283eb4 (GET,200)": 1,
         "https://api.github.com/repos/nodejs/node/git/trees/main (GET,200)": 1,
         "https://api.github.com/repos/nodejs/node/releases (GET,200)": 1,
         "https://api.github.com/repos/nodejs/node/tags (GET,200)": 1,
         "https://api.github.com/repositories/27193779/releases (GET,200)": 3,
         "https://api.github.com/repositories/27193779/tags (GET,200)": 7,
```

(note the last 2 lines)

**Comment by viceice** [comment]
> I note that we still have the link header-based pagination which will cause some private repos to fail:
> 
> ```
>          "https://api.github.com/repos/nodejs/node (GET,200)": 1,
>          "https://api.github.com/repos/nodejs/node/git/blobs/a107bc9c748cf503f04416f780ed362068283eb4 (GET,200)": 1,
>          "https://api.github.com/repos/nodejs/node/git/trees/main (GET,200)": 1,
>          "https://api.github.com/repos/nodejs/node/releases (GET,200)": 1,
>          "https://api.github.com/repos/nodejs/node/tags (GET,200)": 1,
>          "https://api.github.com/repositories/27193779/releases (GET,200)": 3,
>          "https://api.github.com/repositories/27193779/tags (GET,200)": 7,
> ```
> 
> (note the last 2 lines)

no, as we use the token from first page for the others. i recently fixed that.

File: lib/modules/datasource/github-releases/index.ts (hunk 4 above)
Comment (viceice) [reply lib/modules/datasource/github-releases/index.ts:211]: yes, but there are tests for those functions, so coverage should be good. 🤔

**Comment by rarkins** [comment]
What about https://github.com/renovatebot/renovate/pull/17693 ?

File: lib/modules/datasource/github-releases/index.ts (hunk 4 above)
Comment (rarkins) [inline lib/modules/datasource/github-releases/index.ts:211]:
```suggestion
```

File: lib/modules/datasource/github-tags/index.ts (hunk 2 above)
Comment (rarkins) [inline lib/modules/datasource/github-tags/index.ts:76]:
```suggestion
```

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by colinodell** [comment]
@zharinov Is there an issue I can track to know when GraphQL + caching has been implemented for GH datasources?  (We're hitting GH rate limits without caching and are hoping to stay on 32.194.0 until caching is re-added)

**Comment by zharinov** [comment]
> @zharinov Is there an issue I can track to know when GraphQL + caching has been implemented for GH datasources? (We're hitting GH rate limits without caching and are hoping to stay on 32.194.0 until caching is re-added)

@colinodell This one: #18188

**Pull request merged**